### PR TITLE
TES-309: Support for loading GraphDB license in VMSS

### PR DESCRIPTION
**Pull request by mihailradkov**
## Changes

Updated the user data script to authenticate in Azure CLI and download the license.

Added new TF modules:
- configuration - takes care of handling GraphDB configurations and secrets
- identity - responsible for creating and managing a user assigned identity
- vault - creates a key vault for storing sensitive configurations

### CODE REVIEW

File: modules/vm/variables.tf [hunk 1]
```diff
@@ -14,6 +16,8 @@ variable "resource_group_name" {
   type        = string
 }
 
+#
```
Comment (viktor-ribchev) [inline modules/vm/variables.tf:19]: comment ?

File: variables.tf [hunk 2]
```diff
@@ -14,6 +16,14 @@ variable "tags" {
   default     = {}
 }
 
+variable "lock_resources" {
+  description = "Enables a delete lock on the resource group to prevent accidental deletions."
+  type        = bool
+  default     = true
+}
+
+#
```
Comment (viktor-ribchev) [inline variables.tf:25]: Comment?

File: modules/vm/main.tf [hunk 3]
```diff
@@ -123,9 +119,8 @@ resource "azurerm_linux_virtual_machine_scale_set" "graphdb" {
...
 
   os_disk {
-    # TODO: size? caching?
-    caching              = "None"
-    storage_account_type = "Premium_LRS"
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
   }
 
   admin_ssh_key {
```
Comment (viktor-ribchev) [inline modules/vm/main.tf:126]: You can replace var.ssh_key with `file(var.ssh_key)` and provide the file instead of the key as string

**Review by viktor-ribchev — APPROVED** [review]

File: modules/vm/variables.tf (hunk 1 above)
Comment (mihailradkov) [reply modules/vm/variables.tf:19]: Preparation for separating the different groups of configurations

File: modules/vm/main.tf (hunk 3 above)
Comment (mihailradkov) [reply modules/vm/main.tf:126]: :+1: , I will do this in another PR

File: variables.tf (hunk 2 above)
Comment (mihailradkov) [reply variables.tf:25]: Preparation for separating the different groups of configurations

File: main.tf [hunk 4]
```diff
@@ -72,6 +76,46 @@
...
+module "vault" {
+  source = "./modules/vault"
+
+  resource_name_prefix = var.resource_name_prefix
+  resource_group_name  = azurerm_resource_group.graphdb.name
+
+  tags = local.tags
+
+  depends_on = [azurerm_resource_group.graphdb]
+}
```
[the remark this answers is not in the document]
Comment (mihailradkov) [reply main.tf:99]: Security issues will be address in another PR

**Pull request merged**